jwt/__init__.py

```diff
@@ -359,6 +359,17 @@ def load(jwt):
 
 def verify_signature(payload, signing_input, header, signature, key='',
                      verify_expiration=True, leeway=0, **kwargs):
+
+    if isinstance(leeway, datetime.timedelta):
+        try:
+            leeway.total_seconds
```

Comment:
What about replacing the try-catch with 
```
if hasattr(leeway, 'total_seconds') and callable(leeway.total_seconds):
    leeway = leeway.total_seconds()
else:
    leeway = leeway.days * 24 * 60 * 60 + leeway.seconds
```

I think this would avoid overhead caused by the exception handling and be a little clearer to the reader.